File: netcore/Flights/Flights.Core/Services/FlightService.cs
using Flights.Core.Interfaces;
using Flights.Core.Model;
using System.Collections.Generic;

namespace Flights.Core.Services
{
    public class FlightService : IFlightService
    {
        private readonly IFlightRepository _flightRepository;

        public FlightService(IFlightRepository flightRepository)
        {
            _flightRepository = flightRepository;
        }

        public void DeleteFlight(string id)
        {
            _flightRepository.Delete(id);
        }

        public Flight GetFlight(string id)
        {
            return _flightRepository.Get(id);
        }

        public List<Flight> GetFlights(string filter)
        {
            return _flightRepository.GetAll(filter);
        }

        public void InsertFlight(Flight flight)
        {
            _flightRepository.Insert(flight);
        }

        public void UpdateFlight(Flight flight)
        {
            _flightRepository.Update(flight);
        }
    }
}

File: Java/src/main/java/flights/core/services/FlightService.java
package flights.core.services;

import flights.core.interfaces.IFlightRepository;
import flights.core.interfaces.IFlightService;
import flights.core.model.Flight;
import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class FlightService implements IFlightService {

  private IFlightRepository flightRepository;

  public FlightService(IFlightRepository flightRepository){
    this.flightRepository = flightRepository;
  }

  public Flight getFlight(String id) {
    return flightRepository.get(id);
  }

  public List<Flight> getFlights(String filter) {
    return flightRepository.getAll(filter);
  }

  public Flight insertFlight(Flight flight) {
    return flightRepository.insert(flight);
  }

  public Flight updateFlight(Flight flight) {
    return flightRepository.update(flight);
  }

  public void deleteFlight(String id) {
    flightRepository.delete(id);
  }
}
